### ISSUE TITLE
error extracting clusters

### ISSUE THREAD

**Issue by madhubioinfo** [BinPro/CONCOCT#237]

**Comment by madhubioinfo** [comment]
I was extracting clusters from bin and this is my command
 /shared/programs/CONCOCT/scripts/extract_fasta_bins.py  1k_filtered.fa concoct_out_clustering_gt1000.csv --output conco_out


I am getting following error
 Traceback (most recent call last):
   File "extract_fasta_bins.py", line 38, in <module>
       main(args)
         File "extract_fasta_bins.py", line 26, in main
             seqs = [all_seqs[contig_id] for contig_id in contig_ids] 
             KeyError: 1

help me to solve the errors

**Comment by franciscozorrilla** [comment]
Hi Madhu,

I am not part of the CONCOCT group but I may be able to help. I had not realized that they included a script for extracting node FASTA, so I wrote my own using R. I invite you to take a look at it [here](https://github.com/franciscozorrilla/thesis_pipeline/blob/master/multiFASTA2speciesFASTA.R). 
I have not used the CONCOCT script for doing this task, but I imagine it is essentially the same thing. 
I did not write the script with the intention of publishing/sharing it, so it may be a bit cryptic to understand if you are not comfortable with R. To get a better understanding of how the script functions, take a look at how I use it in the parseFASTA rule of this [snakefile ](https://github.com/franciscozorrilla/thesis_pipeline/blob/master/Snakefile) and companion [config file](https://github.com/franciscozorrilla/thesis_pipeline/blob/master/config.yaml). 

Hope it helps!
